mlflow/server/js/src/components/ExperimentPage.js

```diff
@@ -31,75 +34,112 @@ export class ExperimentPage extends Component {
         orderByKey: urlState.orderByKey === undefined ? null : urlState.orderByKey,
         orderByAsc: urlState.orderByAsc === undefined ? true : urlState.orderByAsc === "true",
       },
+      nextPageToken: null,
+      loadingMore: false,
     };
   }
 
+  getExperimentRequestId = getUUID();
+  searchRunsRequestId = getUUID();
+  loadMoreRunsRequestId = getUUID();
+
+  loadData() {
+    const { persistedState, lifecycleFilter } = this.state;
+    const { experimentId } = this.props;
+    const { orderByKey, orderByAsc, searchInput } = persistedState;
+    const orderBy = ExperimentPage.getOrderByExpr(orderByKey, orderByAsc);
+    const viewType = lifecycleFilterToRunViewType(lifecycleFilter);
+
+    this.props.getExperimentApi(experimentId, this.getExperimentRequestId);
+    this.props
+      .searchRunsApi([experimentId], searchInput, viewType, orderBy, this.searchRunsRequestId)
+      .then(this.updateNextPageToken)
+      .catch(Utils.logErrorAndNotifyUser);
```

Comment:
Can we also unset the loadingMore state in this case? The `Utils.logErrorAndNotifyUser` handler seems to work well, it's just that the loadingMore spinner continues to spin because we don't do `this.setState({ nextPageToken, loadingMore: false });`

Reply:
Added reset in catch.